hedera-node/src/main/java/com/hedera/services/txns/contract/ContractDeleteTransitionLogic.java

```diff
@@ -21,81 +21,98 @@
+import com.hedera.services.store.AccountStore;
+import com.hedera.services.store.models.Account;
+import com.hedera.services.store.models.Id;
 import com.hedera.services.utils.EntityNum;
 import com.hedera.services.txns.TransitionLogic;
 import com.hedera.services.txns.validation.OptionValidator;
 import com.hederahashgraph.api.proto.java.ResponseCodeEnum;
 import com.hederahashgraph.api.proto.java.TransactionBody;
-import com.hederahashgraph.api.proto.java.TransactionRecord;
 import com.swirlds.merkle.map.MerkleMap;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
 import javax.inject.Inject;
 import javax.inject.Singleton;
-import java.time.Instant;
 import java.util.function.Function;
 import java.util.function.Predicate;
 import java.util.function.Supplier;
 
+import static com.hedera.services.exceptions.ValidationUtils.validateFalse;
+import static com.hedera.services.utils.EntityIdUtils.asAccount;
 import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.ACCOUNT_EXPIRED_AND_PENDING_REMOVAL;
-import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.FAIL_INVALID;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.OBTAINER_SAME_CONTRACT_ID;
 
 @Singleton
 public class ContractDeleteTransitionLogic implements TransitionLogic {
 	private static final Logger log = LogManager.getLogger(ContractDeleteTransitionLogic.class);
 
-	private final HederaLedger ledger;
-	private final LegacyDeleter delegate;
+	private final AccountStore accountStore;
 	private final OptionValidator validator;
 	private final TransactionContext txnCtx;
+	private final GlobalDynamicProperties dynamicProperties;
 	private final Supplier<MerkleMap<EntityNum, MerkleAccount>> contracts;
 
 	private final Function<TransactionBody, ResponseCodeEnum> SEMANTIC_CHECK = this::validate;
 
 	@Inject
 	public ContractDeleteTransitionLogic(
-			HederaLedger ledger,
-			LegacyDeleter delegate,
+			AccountStore accountStore,
 			OptionValidator validator,
 			TransactionContext txnCtx,
+			GlobalDynamicProperties dynamicProperties,
 			Supplier<MerkleMap<EntityNum, MerkleAccount>> contracts
 	) {
-		this.ledger = ledger;
-		this.delegate = delegate;
+		this.accountStore = accountStore;
 		this.validator = validator;
 		this.txnCtx = txnCtx;
+		this.dynamicProperties = dynamicProperties;
 		this.contracts = contracts;
 	}
 
-	@FunctionalInterface
-	public interface LegacyDeleter {
-		TransactionRecord perform(TransactionBody txn, Instant consensusTime);
-	}
-
 	@Override
 	public void doStateTransition() {
-		try {
-			final var contractDeleteTxn = txnCtx.accessor().getTxn();
-			final var op = contractDeleteTxn.getContractDeleteInstance();
+		/* --- Translate from gRPC types --- */
+		final var contractDeleteTxn = txnCtx.accessor().getTxn();
+		final var op = contractDeleteTxn.getContractDeleteInstance();
+
+		final var targetId = Id.fromGrpcAccount(asAccount(op.getContractID()));
```

Comment:
In the PR for `ContractUpdateTransitionLogic ` we introduced a new static method for generating Id from `ContractID` and line 82 could be like: `final var targetId = Id.fromGrpcContract(op.getContractID());`

This is the implementation:

```
public static Id fromGrpcContract(final ContractID id) {
		return new Id(id.getShardNum(), id.getRealmNum(), id.getContractNum());
	}
```